### PR TITLE
Deprecate normalizeHash method on the rest serializer

### PR DESCRIPTION
**Pull request by bmac**

### CODE REVIEW

File: tests/integration/serializers/rest-serializer-test.js [hunk 1]
```diff
@@ -392,6 +392,29 @@ test('normalizeHash normalizes specific parts of the payload', function(assert)
...
+
+  var jsonHash = {
+    homePlanets: [{ _id: "1", name: "Umber", superVillains: [1] }]
+  };
+
+  run(function() {
+      assert.expectDeprecation(function() {
+        env.restSerializer.normalizeResponse(env.store, HomePlanet, jsonHash, null, 'findAll');
+      }, /`RESTSerializer.normalizeHash` has been deprecated/)
+  });
```
Comment (pangratz) [inline tests/integration/serializers/rest-serializer-test.js]: much indentation, such ocd

**Comment by pangratz** [comment]
LGTM. Let's hope this makes the cut in `3.0` :stuck_out_tongue_winking_eye:

File: tests/integration/serializers/rest-serializer-test.js (hunk 1 above)
Comment (bmac) [inline tests/integration/serializers/rest-serializer-test.js]: Good :eyes:

**Comment by jsangilve** [comment]
@bmac Shouldn't this remove [old tests](https://github.com/bmac/data/blob/deprecate-normalize-hash/tests/integration/serializers/rest-serializer-test.js#L355) relying on `normalizeHash` or should they stay until 3.0?

**Comment by bmac** [comment]
I'd like to leave that test until 3.0 so the `normalizeHash` functionality doesn't regress.

**Pull request merged**